app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/EditRecipeViewModel.kt

```diff
@@ -0,0 +1,219 @@
+        }
+    }
+
+    /**
+     * Updates the recipe's description
+     *
+     * @param recipe The recipe to change
+     * @param description The new description
+     */
+    fun setDescription(recipe: Recipe, description: String) {
+        viewModelScope.launch {
+            editRecipe.setRecipeDescription(recipe, description)
+        }
+    }
+
+    /**
+     * Updates the recipe's picture
+     *
+     * @param recipe The recipe to change
+     * @param uri The new URI for the new picture
+     */
+    fun setRecipePicture(recipe: Recipe, uri: Uri) {
+        viewModelScope.launch {
+            editRecipe.setRecipePicture(recipe, uri)
+        }
+    }
+
+    /**
+     * Updates the number of people the recipe is designed for
+     *
+     * @param recipe The recipe to change
+     * @param numberOfPeople New number of people
+     */
+    fun setNumberOfPeople(recipe: Recipe, numberOfPeople: Int) {
+        viewModelScope.launch {
+            editRecipe.setNumberOfPeople(recipe, numberOfPeople)
+        }
+    }
+
+    /**
+     * Adds a dietary speciality to the recipe
+     *
+     * @param recipe The recipe to change
+     * @param speciality The speciality to add
+     */
+    fun addDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        viewModelScope.launch {
+            editRecipe.addDietarySpeciality(recipe, speciality)
+        }
+    }
+
+    /**
+     * Deletes a speciality from the recipe
+     *
+     * @param recipe The recipe to change
+     * @param speciality The speciality to delete
+     */
+    fun deleteDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        viewModelScope.launch { editRecipe.deleteDietarySpeciality(recipe, speciality) }
+    }
+
+    /**
+     * Adds an ingredient to the given ingredient group, or the whole ingredient group if no ingredient is provided
+     *
+     * @param recipe The recipe to change
+     * @param group The ingredient which should be added, or to which the ingredient is added to
+     * @param ingredient The ingredient to be added ( or null if the ingredient group should be added)
+     */
+    fun addIngredient(recipe: Recipe, group: IngredientGroup, ingredient: Ingredient? = null) {
+        viewModelScope.launch {
+            if (ingredient == null) {
+                editRecipe.addIngredientGroup(recipe, group)
+            } else {
+                editRecipe.addIngredient(recipe, group, ingredient)
+            }
+        }
+
+        /**
+         * Deletes an ingredient group with all its ingredients in it or,
+         * if the ingredient is specified only the ingredient from the group.
+         *
+         * @param recipe The recipe to change
+         * @param group The group to delete or from which the ingredient gets deleted
+         * @param ingredient The ingredient to be deleted ( or null if the ingredient group should be deleted)
+         */
+        fun deleteIngredient(
```

Comment:
This should probably be a member function of the view model and not a local function within ```addInredient()```